storage/swift_storage.py

```diff
@@ -98,8 +99,11 @@ def _download_object_to_file(
         # may need to truncate file if retrying...
         out_file.seek(0)
 
-        resp_headers, object_contents = connection.get_object(
-            container, object_name, resp_chunk_size=_LARGE_CHUNK)
+        try:
+            resp_headers, object_contents = connection.get_object(
+                container, object_name, resp_chunk_size=_LARGE_CHUNK)
+        except ClientException:
```

Comment:
See above comment about S3's `ClientError` and FTP's `error_perm`. The `ClientException` class is generic. We only need to raise `NotFoundError` for a specific case, requiring inspection of the exception object.

Also, chain the original exception.